documentation/src/main/asciidoc/user_guide/persistence.adoc

```diff
@@ -267,33 +348,42 @@ The following is an example configuration for the write-behind strategy:
 </persistence>
 ----
 
+[[segmented_stores]]
 ==== Segmented Stores
+You can configure stores so that data resides in segments to which keys map.
+See link:#key_ownership[Key Ownership] for more information about segments and
+ownership.
+
+Segmented stores increase read performance for bulk operations; for example,
+streaming over data (`Cache.size`, `Cache.entrySet.stream`), pre-loading the
+cache, and doing state transfer operations.
+
+However, segmented stores can also result in loss of performance for write
+operations. This performance loss applies particularly to batch write
+operations that can take place with transactions or write-behind stores. For
+this reason, you should evaluate the overhead for write operations before you
+enable segmented stores. The performance gain for bulk read operations might
+not be acceptable if there is a significant performance loss for write
+operations.
 
-A store can optionally be segmented. That is that data is stored by the segment
-that its key maps to. Information pertaining to segments and the ownership of
-them can be found at link:#key_ownership[Key Ownership].
-
-Segmenting a store has the benefit of increasing the read performance for
-bulk operations. Some examples of said methods are any stream operation
-performed upon the cache, Cache.size as well as preloading and state
-transfer in a cache.
+[IMPORTANT]
+====
+Loss of data can occur if the number of segments in a cache store are not
+changed gracefully. For this reason, if you change the `numSegments` setting in the store configuration, you must migrate the existing store to use the new configuration.
 
-Unfortunately depending on the store implementation, there may be some performance
-loss for write operations. This is especially true for batch writes which can
-occur with transactions or write behind stores, which is mentioned in the previous
-section. Therefore, before enabling segmented stores, this overhead should be
-possibly measured to ensure it is acceptable in turn to gain bulk read performance.
+The recommended method to migrate the cache store configuration is to perform a
+rolling upgrade. The store migrator does not currently support migrating
```

Comment:
I wonder if we should be more explicit, it kinda sounds like it doesn't support it at all currently.

> The store migrator currently supports migrating from a non segmented store to a segmented store, however it does not yet support migrating from a segmented store.

Something like that?